src/07-contact-cards.md

```diff
@@ -0,0 +1,69 @@
+
+The name should be a string sufficiently long as to allow any human name, business name, or entity name.
+
+The avatar should be a URL to an image using an API similar to Gravatar.
+
+Name and avatar are the most common, but it would be good if the protocol were extensible so that services could add other pieces of public profile information (such as website) if such pieces of information were deemed important. The protocol is naturally extensible simply by using JSON.
+
+## Capability discovery
+
+The `.well-known/bsvalias` document is updated to include a contact card endpoint:
+
+```json
+{
+  "bsvalias": "1.0",
+  "capabilities": {
+    "{{fm:brfc}}": "https://example.bsvalias.tld/api/{alias}@{domain.tld}/contact-card"
+  }
+}
+```
+
+The `capabilities.{{fm:brfc}}` is a template URL to query for the contact card information.
+
+## Client Request
+
+The `capabilities.{{fm:brfc}}` path returns a URI template. Senders _MUST_ replace `{alias}`, `{domain.tld}` placeholders with a valid paymail handle.
+
+The client _MUST_ perform a GET request to the obtained URL.
+
+## Server Response
+
+The server response is a JSON object containing contact card information. The only pieces of information are name and avatar. However, because we use JSON, it will be possible to extend this protocol to include additional information later on.
+
+### 200 OK
+
+Returned when a valid request was made. The response _MUST_ have `application/json` as content type. The response body _MUST_ follow this schema:
+
+```json
+{
+  "name":"[a person'a name, a business name, or any other name associated with the paymail]",
+  "avatar":"https://example.com/api/image.png{?s}"
+}
+```
+
+| Field    | Description                                                                                                           |
+|----------|-----------------------------------------------------------------------------------------------------------------------|
+| `name`   | A string up to 100 characters long.                                                                                   |
+| `avatar` | A URL that returns a 80x80 image. It can accept an optional parameter `s` to return an image of width and height `s`. |
+
+The image should be JPEG, PNG, or GIF.
+
+The returned image _MUST_ be a square, at the resolution specified by `s`, and should look good when cropped to a circle, since this will be the most common way to display the images.
+
+Note, however, that there is no way to enforce what resolution image is returned or what the viewer will do with an image that is the incorrect resolution. Thus, if the returned image is not a square or if it is the wrong resolution, the viewer behavior is undefined. Some viewers may crop the image, some may squish the image, and some may choose not to display the image at all.
```

Comment:
I wonder if this last paragraph waters down the standard of what a standards compliant avatar service is and undermines it somewhat. It effectively suggests that application developers have to just deal with non-standards compliant avatar services, and there is no standard just a hope that services will do what is suggested.

What if we proscribed what a standards compliant application developer should do - they still are not required to do it, nor are services required to meet the suggested behaviour, but by doing so it pushes the avatar ecosystem into compliance from the beginning and more or less ensures the quality of how they appear in all our applications by doing so.

1. Avatar urls should only ever return an image of the requested size, or if no size is provided, 80x80.
2. Avatar urls should only ever return images in the JPEG, PNG and GIF formats.
3. If an avatar url returns a non-square image, or a format not listed above, applications should consider the avatar service broken and should not use and should stop requesting their avatars.

Reply:
From the perspective of the web, it is easier to just resize the image than it is to try to detect the size of the image and display something different instead. Thus from our POV we'd rather not have to be bothered with detecting it in the UI. I'm not sure about other languages/platforms.

So here are some options:
- Invalidly sized images are always squished to a square (easiest option for web apps).
- Invalidly sized images are always cropped (second easiest option for web apps).
- Invalidly sized images are detected and not displayed (hardest option for web apps).

The last option does indeed seem to be the most likely to force compliance of square images.

Reply:
I will update the PR with your suggestion of more strongly worded language.